www/pages/[pkg]/[mod]/[symbol].tsx

```diff
@@ -1,186 +1,89 @@
-    typeof params.symbol !== "string"
-  ) {
-    return { notFound: true };
-  }
-  const pkg = docs.getPackage(params.pkg);
-
-  const mod = pkg.modules.find((m) => m.name === params.mod);
-  if (mod === undefined) {
-    return { notFound: true };
-  }
-
-  const find = <T extends { name: string }>(xs: T[]): T | undefined => {
-    return xs.find((x) => x.name === `${mod.name}.${params.symbol as string}`);
-  };
-
-  const sym = coalesceSymbol(
-    {
-      kind: "class",
-      symbol: find(mod.classes),
-    },
-    {
-      kind: "function",
-      symbol: find(mod.functions),
-    },
-    {
-      kind: "variable",
-      symbol: find(mod.variables),
-    }
-  );
-  if (sym === undefined) {
-    return { notFound: true };
-  }
-
-  return {
-    props: {
-      pkg,
-      mod,
-      symbol: sym,
-    },
-  };
-};
-
-export const getStaticPaths: GetStaticPaths = () => {
-  const idx = docs.getPackageIndex();
-
-  const paths: { params: { pkg: string; mod: string; symbol: string } }[] = [];
-  function pushParams(
-    pkg: docs.Pkg,
-    mod: docs.Module,
-    thing: docs.Class | docs.Func | docs.Variable
-  ) {
-    paths.push({
-      params: {
-        pkg: pkg.name,
-        mod: mod.name,
-        symbol: withoutPrefix(mod.name, thing.name),
-      },
-    });
-  }
-
-  idx.forEach((pkg) => {
-    pkg.modules.forEach((mod) => {
-      mod.classes.forEach((cls) => pushParams(pkg, mod, cls));
-      mod.functions.forEach((fn) => pushParams(pkg, mod, fn));
-      mod.variables.forEach((v) => pushParams(pkg, mod, v));
-    });
-  });
-
-  return {
-    paths,
-    fallback: false,
-  };
-};
-
-export default function Page({ pkg, mod, symbol }: Props) {
-  const Sidebar = (() => {
-    switch (symbol.kind) {
-      case "class":
-        return (
-          <>
-            <ClassContents
-              pkg={pkg}
-              mod={mod}
-              cls={symbol.symbol}
-              currentSymbol=""
-            />
-            <ModuleContents pkg={pkg} mod={mod} currentSymbol="" />
-          </>
-        );
-      case "function":
-        return (
-          <ModuleContents
-            pkg={pkg}
-            mod={mod}
-            currentSymbol={symbol.symbol.name}
-          />
-        );
-      case "variable":
-        return (
-          <ModuleContents
-            pkg={pkg}
-            mod={mod}
-            currentSymbol={symbol.symbol.name}
-          />
-        );
-    }
-  })();
+export default function ModulePage() {
+  const router = useRouter();
+  const pkgName = router.query.pkg as string;
+  const modName = router.query.mod as string;
+  const symName = router.query.symbol as string;
+  return (
+    <FetchPackage
+      name={pkgName}
+      content={(pkg) => {
+        const [Sidebar, Content] = (() => {
+          const mod = pkg.modules.find((mod) => mod.name === modName);
+          if (!mod) {
+            return [<ModuleList pkg={pkg} />, `Module ${modName} not found :O`];
```

Comment:
Hmm that string needs DRYing maybe?
```suggestion
            return [<ModuleList pkg={pkg} />, `Module ${modName} not found 🤪`];
```

Reply:
My professional well-considered opinion is:

eh